packages/tools/viewer/src/viewer.ts

```diff
@@ -1078,12 +1113,14 @@ export class Viewer implements IDisposable {
 
         await this._loadModelLock.lockAsync(async () => {
             throwIfAborted(abortSignal, abortController.signal);
-            this._model?.dispose();
-            this._setModel(null);
+            this._activeModel?.dispose();
+            this._setActiveModel(null);
             this.selectedAnimation = -1;
 
             if (source) {
-                this._setModel(await this._loadModel(source, options, abortController.signal), Object.assign({ source, interpolateCamera: false }, options));
+                const model = await this._loadModel(source, options, abortController.signal);
+                this._setActiveModel(model, Object.assign({ source, interpolateCamera: false }, options));
+                this._loadedModelsBacking.push(this._activeModel!);

```

Comment:
Related to previous comment, I don't think we need this change if `_loadModel` adds the model to `_loadedModelsBacking`.